### PR TITLE
workspace folders map should ignore path casing unless linux

### LINKED ISSUE #110188: "Reveal In Side Bar" does not work if the file is opened from external apps
- VSCode Version: 1.51.0
- OS Version: Windows 10 19042

Steps to Reproduce:

1. Open a folder
2. Open a file inside the folder, via File Explorer or via GitHub Desktop
3. See the "Open editors" list shows a full path instead of a relative path
4. Try "Reveal In Side Bar"

image

Does this issue occur when all extensions are disabled?: Yes

I think this is a regression in 1.51.

### PR DESCRIPTION
**Pull request by jrieken**
This PR fixes #110188

### CODE REVIEW

**Review by sandy081 — APPROVED** [review]
LGTM

Note: Did not verify the fix.

**Comment by isidorn** [comment]
While this improves the situation with the `workspace` check. The Explorer still does not reveal. I am investigating further...

**Comment by isidorn** [comment]
Ok, an issue remains, and that is that the `fileService.resolveTo` does not properly resolve when I pass in the wrongly cased drive letter.
@bpasero might you have an idea if the fileService also uses `TernarySearchTree`?

[Code pointer where I use it](https://github.com/microsoft/vscode/blob/59a1893d6d66171437f3608d2e938efa616f421e/src/vs/workbench/contrib/files/common/explorerService.ts#L153)
Since I pass in the wrong cased drive letter the `resolveTo` does not seem to be respected.

To repro on Win 
1. Put a breakpoint in my code pointer
2. Open some workspace. Drag from Windows Explorer a file deep inside that workspace -> it does not get revealed

**Comment by isidorn** [comment]
Ok here it is https://github.com/microsoft/vscode/blob/59a1893d6d66171437f3608d2e938efa616f421e/src/vs/platform/files/common/fileService.ts#L196

Checking if changing it there also helps...

**Comment by isidorn** [comment]
Yeah, that would also have to be changed. I have verified with that change that the issue is fixed.

**Comment by bpasero** [comment]
This code is actually literally copied from how @jrieken had it before I rewrote file service. I think it should use the file system provider's capability for the ternary tree there as well.

**Comment by isidorn** [comment]
@bpasero yeah I just pushed that. Please review and let me know what you think.
If you think it is good I will also push to master.

File: src/vs/platform/files/common/fileService.ts [hunk 1]
```diff
@@ -193,7 +193,7 @@ export class FileService extends Disposable implements IFileService {
 
 			// lazy trie to check for recursive resolving
 			if (!trie) {
-				trie = TernarySearchTree.forUris<true>();
+				trie = TernarySearchTree.forUris<true>(!this.hasCapability(resource, FileSystemProviderCapabilities.PathCaseSensitive));
```
Comment (bpasero) [inline src/vs/platform/files/common/fileService.ts:196]: use `isPathCaseSensitive` in that file

**Review by bpasero — CHANGES_REQUESTED** [review]
Left a comment. Created https://github.com/microsoft/vscode/issues/110241 for the actual solution.

**Comment by jrieken** [comment]
Hm... Should we simply bring back the old behaviour instead of picking on a few places?

**Comment by isidorn** [comment]
@jrieken yeah that might be best.

**Comment by jrieken** [comment]
closing and creating another PR which basically undoes my initial refactoring

**Pull request closed without merging**